apps/admin-gui/src/app/shared/components/dialogs/remove-resource-dialog/remove-resource-dialog.component.ts

```diff
@@ -36,15 +36,19 @@ export class RemoveResourceDialogComponent implements OnInit {
   }
 
   onSubmit() {
-    // TODO Removes only one resource at the time. In future there would be need to remove more than one resource.
     this.loading = true;
-    this.resourcesService.removeResource(this.data.resources[0].id).subscribe( () => {
+    if (this.data.resources.length === 0) {
       this.translate.get('DIALOGS.REMOVE_RESOURCES.SUCCESS').subscribe(successMessage => {
         this.loading = false;
         this.notificator.showSuccess(successMessage);
         this.dialogRef.close(true);
       });
-    });
+    } else {
+      this.resourcesService.removeResource(this.data.resources[0].id).subscribe( () => {
```

Comment:
You need to handle here also the possible error. If the deletion of some resource fails, it doesn't stop loading and it doesn't close the dialog.

Reply:
well if something fails, the error notification is shown( that is something that we want)
then it complete deleting other resources (that is also something we want)
so when the error occurs you want to be loading = false or do you want to close the dialog?

Reply:
You are wrong. If some call fails, this method will not be called again and it will do nothing (except showing the notification). It will not try to delete other resources and the spinner will keep spinning. If it fails, you should close the dialog and refresh the table. This way the user will see updated data if some of the resources were actually deleted.